src/main/java/seedu/address/logic/parser/ListCommandParser.java

```diff
@@ -0,0 +1,32 @@
+package seedu.address.logic.parser;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import seedu.address.logic.commands.ListCommand;
+import seedu.address.model.deck.CardPerformanceMatchesPerformancesPredicate;
+import seedu.address.model.deck.Performance;
+
+/**
+ * Parses input elements and creates a new ListCommand object
+ */
+public class ListCommandParser implements ParserInterface<ListCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the ListCommand
+     * and returns an ListCommand object for execution.
+     */
+    public ListCommand parse(String args) {
+        String trimmedArgs = args.trim();
+        if (!trimmedArgs.isEmpty()) {
+            String[] keywords = trimmedArgs.split("\\s+");
+            Set<Performance> performanceSet =
+                    Arrays.stream(keywords).filter(Performance::isValidPerformance)
```

Comment:
I like this segment - the use of functional programming style makes the code elegant and clean